### DISCUSSION TITLE
Klipper Enabled Ender 3 - Teddy Warner

### DISCUSSION THREAD

**Discussion by giscus** [Twarner491/theodore.net#8] [General]
# Klipper Enabled Ender 3 - Teddy Warner

Project Portfolio and Documentation Site for Teddy Warner - A Student, Maker, Developer, Award Winning Designer & Graduate of the Fab Academy.

https://teddywarner.org/Projects/KlipperEnder3/

**Comment by Gabed699** [comment]
Hi I have two questions is your main board a V4 board or older and how you connected your pi to your mainboard via usb or alternate method. Thx

**Reply by Twarner491** [reply]
Hello! 

The main board I've used on my machine is the *BTT SKR Mini E3 V2*, however the Klipper installation process outlined in this article will be the same for the V4 main board. My RPI and main board are connected with a micro USB cable (which is definitely the simplest option), yet there are plenty of documented methods utilizing RPI GPIO pins to communicate with a main board, possibly providing a cleaner alternative to a USB cable. [Here](https://www.reddit.com/r/klippers/comments/mtczj8/connecting_to_klipper_on_mini_e3_v2_through_gpio/) is a thread I found with some additional details regarding alternative connection options between RPI and main board - **if there's enough interest in these alternatives, I'll parse though the best options and update this article**.

Best of luck to you with your projects!
